sdk/include/opentelemetry/sdk/resource/resource.h

```diff
@@ -18,8 +21,7 @@ namespace sdk
 namespace resource
 {
 
-using ResourceAttributes =
-    std::unordered_map<std::string, opentelemetry::sdk::common::OwnedAttributeValue>;
+using ResourceAttributes = opentelemetry::sdk::common::AttributeMap;
```

Comment:
Switching to "smarter" map that performs transform of incoming `const char *` values to `std::string`.